packages/extension-core/src/domains/nfts/service.ts

```diff
@@ -0,0 +1,172 @@
+    const accounts = getNftsAccountsList()
+
+    const requiresUpdateAccount = accounts.some((account) => !prevAccounts.includes(account))
+
+    const requiresUpdateTimestamp =
+      !nftsStore.value.timestamp || Date.now() - nftsStore.value.timestamp >= UPDATE_INTERVAL
+
+    if (!requiresUpdateAccount && !requiresUpdateTimestamp) return
+
+    status.next("loading")
+    promise = updateData()
+      .then(() => {
+        status.next("loaded")
+        errorsStreak = 0
+      })
+      .catch(() => {
+        status.next("stale")
+        errorsStreak++
+        if (errorsStreak >= 3) stop = true
+      })
+      .finally(() => {
+        promise = null
+      })
+  }
+
+  const interval = setInterval(checkOrUpdate, UPDATE_CHECK_INTERVAL)
+
+  checkOrUpdate()
+
+  const unsubscribe = () => {
+    watcher = null
+    stop = true
+    clearInterval(interval)
+  }
+
+  return unsubscribe
+}
+
+subscriptions.subscribe(async (subIds) => {
+  if (subIds.length === 0 && watcher) {
+    try {
+      watcher.then((unsubscribe) => unsubscribe())
+    } catch (err) {
+      log.error("Error unsubscribing from nft data", { err })
+    }
+  } else if (subIds.length && !watcher) {
+    watcher = watchData()
+  }
+})
+
+const obsNfts = combineLatest([nftsStore, status]).pipe(
+  map(([store, status]) => {
+    const { collections, nfts, timestamp, favoriteNftIds, hiddenNftCollectionIds } = store
+    const data: NftData = {
+      collections,
+      nfts,
+      timestamp,
+      status,
+      favoriteNftIds,
+      hiddenNftCollectionIds,
+    }
+    return data
+  })
+)
+
+export const subscribeNfts = (callback: (data: NftData) => void) => {
+  const subscription = obsNfts.subscribe((next) => {
+    callback(next)
+  })
+
+  const id = addSubscription()
+
+  return () => {
+    removeSubscription(id)
+    subscription.unsubscribe()
+  }
+}
+
+export const setHiddenNftCollection = (id: string, isHidden: boolean) => {
+  const hiddenNftCollectionIds = nftsStore.value.hiddenNftCollectionIds.filter((cid) => cid !== id)
+  if (isHidden) hiddenNftCollectionIds.push(id)
+
+  nftsStore.next({
+    ...nftsStore.value,
+    hiddenNftCollectionIds,
+  })
+}
+
+export const setFavoriteNft = (id: string, isFavorite: boolean) => {
+  const favoriteNftIds = nftsStore.value.favoriteNftIds.filter((nid) => nid !== id)
+  if (isFavorite) favoriteNftIds.push(id)
+
+  nftsStore.next({
+    ...nftsStore.value,
+    favoriteNftIds,
+  })
+}
+
+export const refreshNftMetadata = async (id: string) => {
+  const nft = nftsStore.value.nfts.find((n) => n.id === id)
+  if (!nft) return
+
+  const { evmNetworkId, contract, tokenId } = nft
+
+  await fetchRefreshNftMetadata(evmNetworkId, contract.address, tokenId)
+
+  // force an update after 10 seconds, might be lucky !
+  await sleep(10_000)
+  updateData()
+
+  // we don't know when the refresh will be done, lower the update interval to 10 minute for this session
+  UPDATE_INTERVAL = 60 * 1000
```

Comment:
Do we have any way of knowing when the update was successful, so we can reset UPDATE_INTERVAL to a longer time?